symengine/sets.cpp

```diff
@@ -90,6 +92,19 @@ RCP<const Set> Interval::close() const
     return interval(start_, end_, false, false);
 }
 
+bool Interval::contains(const RCP<const Basic> &a) const
```

Comment:
It fails.
Because we don't know the `min` of `(sqrt(i2), zero)` and `max` of `(sqrt(i2), im5)`.
Should we throw an "Not implemented error" for `Basic` ?